src/slice.rs

```diff
@@ -143,35 +144,39 @@ impl<'a, T> NonEmptyIterator for NEChunks<'a, T> {
 
     fn next(&mut self) -> Option<Self::Item> {
         if self.index == 0 {
-            let end = self.window.get() - 1;
+            let end = min(self.window.get() - 1, self.tail.len());
 
             let slice = NESlice {
                 head: self.head,
                 tail: &self.tail[0..end],
             };
 
-            self.index = end;
+            // 2024-03-18: This is a workaround for edge case of 1 element slice, thus the index
+            // does not get incremented as end is 0, thus the next iteration will return the same
+            // slice gain and again.
+            self.index = end + 1;
```

Comment:
Since we do not allow to have the index starting at -1, this is a trick to avoid endless looping on the iterator when we have a single element on the collection.
Obviously this has an impact on the comparison with the tail length for the stop case.